queues.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+// A queue must be read and write
+type Queue interface {
+	send(DomainEntry) error
+	pull(int) ([]DomainEntry, error)
+	peek(string) int
+	persist() error
+}
+
+type cachedQueue struct {
+	store   map[string]int
+	backend Backend
+}
+
+func NewCachedQueue(persistanceBackend Backend) *cachedQueue {
+	q := new(cachedQueue)
+	q.store = make(map[string]int)
+	q.backend = persistanceBackend
+	return q
+}
+
+func (q cachedQueue) send(entry DomainEntry) error {
+	stats := q.store[entry.Domain]
+	if stats == -1 {
```

Comment:
What does `-1` mean here?

Reply:
Good catch! Magic numbers are very unmagical.

A -1 means that it previously bounced.

Reply:
Yeah, this part of the code is quite confusing to be honest.

Reply:
I will clarify it shortly. 👍